api/client.go

```diff
@@ -99,17 +103,51 @@ func NewClient(l logger.Logger, conf Config) *Client {
 	}
 
 	return &Client{
-		logger: l,
-		client: agenthttp.NewClient(clientOptions...),
-		conf:   conf,
+		logger:         l,
+		client:         agenthttp.NewClient(clientOptions...),
+		conf:           conf,
+		requestHeaders: requestHeadersFromEnv(os.Environ()),
```

Comment:
For sub-processes running inside job commands like `buildkite-agent annotate`, this is where we grab `BUILDKITE_REQUEST_HEADER_BUILDKITE_…` from environment into `Buildkite-*` request headers.